### PR TITLE
Allow safe decompression inside vehicles (configurable)

### PR DESCRIPTION
**Pull request by joker-119**
Update references to be stored locally in the repo (I seriously don't know why more people don't do this, it infuriates me).

Allow safe decompression inside of vehicles (Seamoth, PRAWN, Cyclops) via toggleable configuration.

### CODE REVIEW

**Comment by Alexejhero** [comment]
Everything allright except for the fact that you included `Assembly-CSharp.dll` and `Assembly-CSharp-firstpass.dll`. Those are game files and should not be uploaded to GitHub.

**Comment by Alexejhero** [comment]
Removing them wont do the trick, you actually need to rewrite the git commit.

@galaxy119 Make sure you have the "Allow edits from maintainers" option enabled and I'll do it for you

**Comment by PrimeSonic** [comment]
> Everything allright except for the fact that you included `Assembly-CSharp.dll` and `Assembly-CSharp-firstpass.dll`. Those are game files and should not be uploaded to GitHub.

> Removing them wont do the trick, you actually need to rewrite the git commit.
> 
> @galaxy119 Make sure you have the "Allow edits from maintainers" option enabled and I'll do it for you



@seraphimrisen 
You can see a similar pattern on mine and other repos.
https://github.com/PrimeSonic/PrimeSonicSubnauticaMods/tree/master/DependenciesUnity5
https://github.com/PrimeSonic/PrimeSonicSubnauticaMods/tree/BelowZeroConversions/DependenciesUnity2018
Having the DLLs copied to a common folder inside the repo itself does make collaboration easier.

But, like @AlexejheroYTB said, this should be limited to the Unity and mod DLLs.
The Assembly-CSharp DLLs are game code and shouldn't be getting redistributed.

File: NitrogenMod/Items/ReinforcedSuitsCore.cs [hunk 1]
```diff
@@ -52,15 +52,15 @@ protected ReinforcedSuitsCore(string classID, string friendlyName, string descri
...
+            GameObject prefab = CraftData.GetPrefabForTechType(BaseType);
             var obj = GameObject.Instantiate(prefab);
 
             return obj;
         }
 
         private void SetEquipmentType()
         {
-            CraftDataHandler.SetEquipmentType(this.TechType, this.DiveSuit);
+            CraftDataHandler.SetEquipmentType(TechType, DiveSuit);
```
Comment (PrimeSonic) [inline NitrogenMod/Items/ReinforcedSuitsCore.cs]: Changes like this (and there are _many_ throughout this pull request) are just code style change which obfuscate exactly what meaningful change you're trying to make.

Whatever your personal feels are about where `using` statements or whether `this` should precede a property or not is irrelevant.
A pull request that is 90% code style change and 10% actual change is unhepful.

I would sincerely recommend either starting over: removing the Assembly-CSharp DLLs and reverting any non-functional changes like in this file.

Comment (Alexejhero) [reply NitrogenMod/Items/ReinforcedSuitsCore.cs]: This could be just Visual Studio automatically removing these based on styling settings.

Comment (joker-119) [reply NitrogenMod/Items/ReinforcedSuitsCore.cs]: It's indeed Jetbrains just auto-applying that style setting.

**Pull request closed without merging**